community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CommunityAdministrationCommandRuntime.scala

```diff
@@ -211,8 +211,8 @@ case class CommunityAdministrationCommandRuntime(normalExecutionEngine: Executio
           }
           .handleResult((_, value, p) => {
             val oldCredentials = SystemGraphCredential.deserialize(value.asInstanceOf[TextValue].stringValue(), secureHasher)
-            val newValue = p.get(newPw.bytesKey).asInstanceOf[ByteArray].asObject()
-            val currentValue = p.get(currentKeyBytes).asInstanceOf[ByteArray].asObject()
+            val newValue = p.get(newPw.bytesKey).asInstanceOf[ByteArray].asObjectCopy()
```

Comment:
We cannot use `asObjectCopy()` here. We do not want to keep extra copies of passwords in memory, since we zero the bytes after the command has been executed (line 231).

Reply:
Reverted, thanks.